server/admin_management/api/app/services/fam_admin_user_access_service.py

```diff
@@ -0,0 +1,191 @@
+        is_user_fam_admin = len(list(filter(
+            lambda grant: grant.application_name == APPLICATION_FAM,
+            user_admin_privilege))) != 0
+        if is_user_fam_admin:
+            user_access_grants.append(self.__construct_fam_admin_auth_grant())
+
+        is_user_app_admin = len(user_apps_admin_privilege) != 0
+        if is_user_app_admin:
+            user_access_grants.append(self.__construct_app_admin_auth_grant(
+                user_apps_admin_privilege
+            ))
+
+        is_user_delegated_admin = len(user_delegated_admin_privilege) != 0
+        if is_user_delegated_admin:
+            user_access_grants.append(
+                self.__construct_deldgated_admin_auth_grant(
+                    user_delegated_admin_privilege)
+            )
+
+        # final response constructed for list of access grants.
+        admin_user_access_response = FamAdminUserAccessResponse(**{
+            "access": user_access_grants
+        })
+
+        return admin_user_access_response
+
+    # -- Private methods
+
+    def __construct_fam_admin_auth_grant(self):
+        """
+        Constract (auth_key = FAM_ADMIN) access grants.
+        """
+        fam_applications = self.application_repo.get_applications()
+        fam_admin_auth_grant = FamAuthGrantDto(**{
+            "auth_key": AdminRoleAuthGroup.FAM_ADMIN,
+            "grants": self.__preprocess_grant_details(list(map(
+                lambda fam_application: FamGrantDetailDto(**{
+                    "application": FamApplicationDto(
+                        **fam_application.__dict__
+                    )
+                }), fam_applications)))
+        })
+
+        return fam_admin_auth_grant
+
+    def __construct_app_admin_auth_grant(
+        self,
+        granted_applications: List[FamApplication]
+    ):
+        """
+        Constract (auth_key = APP_ADMIN) access grants.
+        """
+        app_admin_auth_grant = FamAuthGrantDto(**{
+            "auth_key": AdminRoleAuthGroup.APP_ADMIN,
+            "grants": self.__preprocess_grant_details(list(map(
+                lambda fam_application: FamGrantDetailDto(**{
+                    "application": FamApplicationDto(
+                        **fam_application.__dict__
+                    ),
+                    "roles": list(map(
+                        lambda role: FamRoleDto(**role.__dict__),
+                        self.role_repo.get_base_roles_by_app_id(
+                            fam_application.application_id
+                        )
+                    ))
+                }), granted_applications)))
+        })
+
+        return app_admin_auth_grant
+
+    def __construct_deldgated_admin_auth_grant(
+        self,
+        granted_roles: List[FamRole]
+    ):
+        """
+        Constract (auth_key = DELEGATED_ADMIN) access grants.
+        """
+        delegated_admin_grants_details = []
+        app_grouped_granted_roles = \
```

Comment:
shall we add a comment about what variable looks like? an example value maybe?

Reply:
Yeah, I was thinking the same things but I felt too much comments on code to explain might get very destructive.
I will try put some since you feel so as well.

Reply:
A bit of comment explanation added. See if that's clear to you. Thanks!